Templatereplacer: fix files input namespace + add tests

ramirezfranciscof opened the pull request:
The `files` input namespace of the `TemplatereplacerCalculation` was not dynamic (so it was basically an unusable namespace with no inputs in it). This is a quick fix for that. Additionally, I have added a test for all of its basic functionalities and a test specifically for said namespace.

sphuber commented:
@ramirezfranciscof I think you may have used an incorrect branch because this seems to include two unrelated commits.

ramirezfranciscof commented:
@sphuber There, I fixed that, rebased and all test pass. Thanks for the heads up, I promise I will eventually make a good PR from the first push (⌒_⌒;)

sphuber commented on tests/calculations/test_templatereplacer.py:
```diff
@@ -0,0 +1,113 @@
+def test_file_usage(fixture_sandbox, aiida_localhost, generate_calc_job, tmpdir):
+    """Test a base template that uses two files."""
+
+    file1_content = 'Content of file 1'
+    file1_handle = tmpdir.join('file1')
+    file1_handle.write(file1_content)
+    file1_node = orm.SinglefileData(file=file1_handle.strpath)
```
```suggestion
    file1_node = orm.SinglefileData(io.StringIO('Content of file 1'))
```

sphuber commented on tests/calculations/test_templatereplacer.py:
```diff
@@ -0,0 +1,113 @@
+    file1_handle.write(file1_content)
+    file1_node = orm.SinglefileData(file=file1_handle.strpath)
+
+    file2_content = 'Content of file 2'
+    file2_handle = tmpdir.join('file2')
+    file2_handle.write(file2_content)
+    file2_node = orm.SinglefileData(file=file2_handle.strpath)
```
```suggestion
file2_node = orm.SinglefileData(io.StringIO('Content of file 2'))
```

sphuber reviewed (changes requested):
Thanks @ramirezfranciscof just a minor simplification

sphuber commented:
Sorry @ramirezfranciscof that should have been `io.BytesIO(b'some_string')`. I will correct it. Anyway need to rebase it

ramirezfranciscof commented:
Ah, no problem, I didn't notice either! For me its ready to merge.
